modules/ssl/ssl_engine_pphrase.c

```diff
@@ -118,6 +118,16 @@ static apr_file_t *writetty = NULL;
 static apr_file_t *readtty = NULL;
 
 int ssl_pphrase_Handle_CB(char *, int, int, void *);
+#if defined(HAVE_OPENSSL_ENGINE_H) && defined(HAVE_ENGINE_INIT)
```

Comment:
As getting the password from app to the engine is a separate issue to adding pkcs11 URI support, it may be a good idea to separate the patches. That would be easier for upstream to review, and also understand that they are two different issues.

Reply:
Ok, I will separate the changes in different patches.